File: cpp/Powersupply.cpp
// PowerSupply.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
using namespace std;

class PowerSupply : public Komponen {
private:
    int wattage;
    string certification;
    bool modular;
    
public:
    PowerSupply() {}
    
    PowerSupply(int wattage, string certification, bool modular, string merk, string nama) : Komponen(merk, nama) {
        this->wattage = wattage;
        this->certification = certification;
        this->modular = modular;
    }
    
    void setWattage(int wattage) {
        this->wattage = wattage;
    }
    
    void setCertification(string certification) {
        this->certification = certification;
    }
    
    void setModular(bool modular) {
        this->modular = modular;
    }
    
    int getWattage() {
        return this->wattage;
    }
    
    string getCertification() {
        return this->certification;
    }
    
    bool getModular() {
        return this->modular;
    }
    
    ~PowerSupply() {}
};

File: python/PowerSupply.py
from Komponen import Komponen
class PowerSupply(Komponen) :
    def __init__(self, wattage, certification, modular, merk, nama):
        super().__init__(merk, nama)
        
        self.__wattage = wattage
        self.__certification = certification
        self.__modular = modular
        
    def get_wattage(self):
        return self.__wattage
    def get_certification(self):
        return self.__certification
    def get_modular(self):
        return self.__modular
    
    # Setter methods
    def set_wattage(self, wattage):
        self.__wattage = wattage
    def set_certification(self, certification):
        self.__certification = certification
    def set_modular(self, modular):
        self.__certification = modular
        